core/go/internal/domainmgr/domain.go

```diff
@@ -86,7 +86,7 @@ func (dm *domainManager) newDomain(name string, conf *pldconf.DomainConfig, toDo
 		dm:              dm,
 		conf:            conf,
 		defaultGasLimit: pldconf.DefaultDefaultGasLimit, // can be set by config below
-		initRetry:       retry.NewRetryIndefinite(&conf.Init.Retry),
+		initRetry:       retry.NewRetryLimited(&conf.Init.Retry),
```

Comment:
This is a very significant functional change. Really need a clear description of everywhere this is used, and why a non-infinite retry is safe.

Reply:
By default, the `Limited` parameter is set to `0` [when no explicit value is provided](https://github.com/LF-Decentralized-Trust-labs/paladin/blob/c1b5160704306e8529e1a59f261d42f3e7a35a74/toolkit/go/pkg/retry/retry.go#L55C2-L55C75), which is then [interpreted as infinite retries](https://github.com/LF-Decentralized-Trust-labs/paladin/blob/c1b5160704306e8529e1a59f261d42f3e7a35a74/toolkit/go/pkg/retry/retry.go#L70C31-L70C48). Therefore, this change does not affect the current behavior. As I explained in the PR, the reason...

Reply:
Going to take a look and understand the race condition in the test - we do need a test-only change here.

Reply:
I've proposed a commit directly to the branch, that:
- Updates some of the missing code comments that lead to confusion
- Splits the `DomainRegistered` function so the unit tests can call `UTSetMaxAttempts(1)` before the init() routine kicks off

Hope that addresses the timing condition in the test 🤞